skyportal/handlers/api/internal/source_views.py

```diff
@@ -0,0 +1,50 @@
+import datetime
+from sqlalchemy import func, desc
+from baselayer.app.access import permissions, auth_or_token
+from ...base import BaseHandler
+from ....models import (
+    DBSession, Source, SourceView, GroupSource
+)
+
+
+class SourceViewsHandler(BaseHandler):
+    @auth_or_token
+    def get(self):
+        prefs = (self.current_user.preferences if
+                 hasattr(self.current_user, 'preferences') and
+                 self.current_user.preferences else {})
+        if 'topSources' in prefs and 'maxNumSources' in prefs['topSources']:
```

Comment:
I think we discussed the possibility of a defaults dictionary that gets updated. Thoughts?

Reply:
Can you explain what you're thinking here? Can we set up our JSONB column to be a default dict? I'm a bit unclear about what you mean

Reply:
I mean that, somewhere in the app, on frontend or backend, we should define the default values for the preferences.  And instead of having logic for each branch, just start off with the default values dictionary, and update it with whatever is set in the preferences.

Reply:
How about adding a default value in the model definition?